src/lib/y2storage/proposal/autoinst_devices_creator.rb

```diff
@@ -89,27 +88,18 @@ def populated_devicegraph(planned_devices, disk_names)
         # Process planned disk like devices (Xen virtual partitions and full disks)
         planned_disk_like_devs = process_disk_like_devs(planned_devices, creator_result.devicegraph)
 
+        devs_to_create = parts_to_create
+
         # Add planned disk like devices to reuse list so they can be considered for lvm and raids
         # later on.
         devs_to_reuse = parts_to_reuse + planned_disk_like_devs
 
-        # Process planned Mds
-        mds_to_create, mds_to_reuse, creator_result =
-          process_mds(planned_devices, devs_to_reuse, creator_result)
-        devs_to_reuse.concat(mds_to_reuse.flat_map(&:partitions))
-
-        # Process planned bcaches
-        bcaches_to_create, bcaches_to_reuse, creator_result =
-          process_bcaches(planned_devices, devs_to_reuse, creator_result)
-        devs_to_reuse.concat(bcaches_to_reuse.flat_map(&:partitions))
+        creator_result = add_mds(planned_devices, devs_to_create, devs_to_reuse, creator_result)
+        creator_result = add_bcaches(planned_devices, devs_to_create, devs_to_reuse, creator_result)
+        creator_result = add_vgs(planned_devices, devs_to_create, devs_to_reuse, creator_result)
+        creator_result = add_nfs_filesystems(planned_devices, devs_to_create, creator_result)
```

Comment:
what is reason to also return that creator_result and not having it as rw parameter only? I am not sure if this is more readable.

Reply:
And also a bit confuse me as for me it looks like creator result is not modified. Is it modified or not ( I mean third parameter).

Reply:
Well, first of all, I did this change only because rubocop was complaining because too many assignments. I agree this solution is quite confusing. The last parameter (creator_result) is not modified inside the method, for this reason a new creator_result is always returned. I will check if it is possible to modify it inside each method.